From 1dcf087af57a93ec7f24aa8be73e7ff0ffad22ca Mon Sep 17 00:00:00 2001
From: Artem Senichev <a.senichev@yadro.com>
Date: Tue, 13 Mar 2018 15:03:41 +0300
Subject: [PATCH] Fix POS and SPD-VPD_REC_NUM

Hank Chang: When bringup this 4Socket 128DIMMs system, we hit the following
VPD/SPD issue and suspected it might be related to the VPD_REC_NUM numbering
algorithm issue then caused the SPD cache overlap for the 4 sockets case.
So we simply tried to jump the $aff_pos offset from 16 to 64 for each Proc and
the issue was fixed/gone.

Originally created by MSI (S188)

Signed-off-by: Artem Senichev <a.senichev@yadro.com>
---
 src/usr/targeting/common/Targets.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/usr/targeting/common/Targets.pm b/src/usr/targeting/common/Targets.pm
index 35f08428c..4eaba16e7 100644
--- a/src/usr/targeting/common/Targets.pm
+++ b/src/usr/targeting/common/Targets.pm
@@ -744,7 +744,7 @@ sub processMcs
                         $self->setAttribute($dimm,"MBA_PORT",$port_num);
                         $self->setAttribute($dimm,"MBA_DIMM",$dimm_num);
 
-                        my $aff_pos=16*$proc+$mcs*$self->{MAX_MCS}+4*$mba+
+                        my $aff_pos=64*$proc+$mcs*$self->{MAX_MCS}+4*$mba+
                                     2*$port_num+$dimm_num;
                         $self->setAttribute($dimm, "AFFINITY_PATH",
                             $parent_affinity
-- 
2.14.1

